Modules/TOF/src/TaskTOFPID.cxx

```diff
@@ -0,0 +1,298 @@
+
+  if (mRecoCont.isTrackSourceLoaded(GID::ITSTPCTOF)) { // if track is ITS+TPC
+
+    mITSTPCTracks = mRecoCont.getTPCITSTracks();
+    mITSTPCTOFMatches = mRecoCont.getITSTPCTOFMatches();
+    mTPCTracks = mRecoCont.getTPCTracks();
+
+    mMyTracks.clear();
+
+    // loop over TOF MatchInfo
+    for (const auto& matchTOF : mITSTPCTOFMatches) {
+
+      GTrackID gTrackId = matchTOF.getTrackRef();
+      const auto& trk = mITSTPCTracks[gTrackId.getIndex()];
+      const auto& trkTPC = mTPCTracks[trk.getRefTPC()];
+
+      const o2::track::TrackLTIntegral& info = matchTOF.getLTIntegralOut();
+
+      if (!selectTrack(trkTPC)) { // Discard tracks according to configurable cuts
+        continue;
+      }
+
+      mMyTracks.push_back(MyTrack(matchTOF, trk));
+
+    } // END loop on TOF matches
+
+    std::vector<MyTrack> tracks;
+
+    // sorting matching in time
+    std::sort(mMyTracks.begin(), mMyTracks.end(),
+              [](MyTrack a, MyTrack b) { return a.tofSignalDouble() < b.tofSignalDouble(); });
+
+    for (int i = 0; i < mMyTracks.size(); i++) { // loop looking for interaction candidates
+      tracks.clear();
+      int ntrk = 1;
+      double time = mMyTracks[i].tofSignalDouble();
+      tracks.emplace_back(mMyTracks[i]);
+      for (; i < mMyTracks.size(); i++) {
+        double timeCurrent = mMyTracks[i].tofSignalDouble();
+        if (timeCurrent - time > 100E3) {
+          i--;
+          break;
+        }
+        tracks.emplace_back(mMyTracks[i]);
+        ntrk++;
+      }
+      if (ntrk > 0) { // good candidate with time
+        processEvent(tracks);
+      }
+    }
+  } // END if track is ITS-TPC
+
+  ILOG(Info, Support) << " Processed! " << ENDM;
+  return;
+}
+
+void TaskTOFPID::endOfCycle()
+{
+
+  ILOG(Info, Support) << "endOfCycle" << ENDM;
+}
+
+void TaskTOFPID::endOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "endOfActivity" << ENDM;
+}
+
+void TaskTOFPID::reset()
+{
+  // clean all the monitor objects here
+  mHistDeltatPi->Reset();
+  mHistDeltatKa->Reset();
+  mHistDeltatPr->Reset();
+  mHistDeltatPiPt->Reset();
+  mHistDeltatKaPt->Reset();
+  mHistDeltatPrPt->Reset();
+  mHistMass->Reset();
+  mHistBetavsP->Reset();
+}
+
+void TaskTOFPID::processEvent(const std::vector<MyTrack>& tracks)
+{
+
+  auto evtime = o2::tof::evTimeMaker<std::vector<MyTrack>, MyTrack, MyFilter>(tracks);
+
+  int nt = 0;
+  for (auto& track : tracks) {
+
+    float mT0 = evtime.eventTime;
+    float mT0Res = evtime.eventTimeError;
+    //
+    float sumw = 1. / (mT0Res * mT0Res);
+    mT0 *= sumw;
+    mT0 -= evtime.weights[nt] * evtime.tracktime[nt];
+    sumw -= evtime.weights[nt];
+    mT0 /= sumw;
+    mT0Res = sqrt(1. / sumw);
+
+    nt++;
+
+    // Beta
+    float beta = track.getL() / (track.tofSignal() - mT0) * cinv;
+    // Mass
+    float mass = track.getP() / beta * TMath::Sqrt(TMath::Abs(1 - beta * beta));
+
+    mHistDeltatPi->Fill(track.tofSignal() - mT0 - track.tofExpSignalPi());
```

Comment:
Can you define a variable
tofsignal - to - exp
and fill all histo with that instead of redoing the sum every time?